typescript/infra/scripts/verify-validator.ts

```diff
@@ -104,170 +89,259 @@ class S3Wrapper {
   }
 }
 
-async function main() {
-  const {
-    a: _validatorAddress,
-    p: prospectiveBucket,
-    c: controlBucket,
-  } = await getArgs();
+class Validator {
+  private readonly controlS3BucketClient: S3Wrapper;
+  private readonly prospectiveS3BucketClient: S3Wrapper;
 
-  const cClient = new S3Wrapper(controlBucket);
-  const pClient = new S3Wrapper(prospectiveBucket);
+  // accumulators for stats
+  private extraCheckpoints!: number[];
```

Comment:
nit: why is the `!` necessary? can we call `initStatsState` in the constructor and remove?

Reply:
Because if you run validate twice, ideally it would work both times. If I init in the constructor that would no longer hold true.

Reply:
validate calls `initStatsState` also?